sql-plugin/src/main/scala/com/nvidia/spark/rapids/SpillableColumnarBatch.scala

```diff
@@ -117,8 +117,8 @@ class SpillableColumnarBatchImpl (
   }
 
   override def getColumnarBatch(): ColumnarBatch = {
+    GpuSemaphore.acquireIfNecessary(TaskContext.get())
```

Comment:
Why? Is this because we can have locked the rapids buffer while blocked waiting to get on the semaphore? Did you see this show up in practice?

This feels like a bug fix and not a part of reparation-based fallback. I would much rather see this as a separate issue/PR.

Reply:
This piece of code is irrelavant to repartition based agg, so it will be removed by a subsequent commit

Reply:
This piece of code is irrelavant to repartition based agg, so it will be removed by a subsequent commit